src/main/kotlin/no/nav/familie/ef/sak/vedtak/VedtakHistorikkBeregner.kt

```diff
@@ -1,54 +1,145 @@
+        return this.copy(datoFra = datoFra)
+    }
+
+    override fun medTil(datoTil: LocalDate): Vedtakshistorikkperiode {
+        return this.copy(datoTil = datoTil)
+    }
+}
+
 object VedtakHistorikkBeregner {
 
-    fun lagVedtaksperioderPerBehandling(vedtaksliste: List<Vedtak>,
-                                        datoOpprettetPerBehandling: Map<UUID, LocalDateTime>): Map<UUID, List<Vedtaksperiode>> {
+    private val beregningBarnetilsyn: BeregningBarnetilsynService = BeregningBarnetilsynService()
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun lagVedtaksperioderPerBehandling(vedtaksliste: List<AndelHistorikkBeregner.BehandlingHistorikkData>,
+                                        datoOpprettetPerBehandling: Map<UUID, LocalDateTime>): Map<UUID, List<Vedtakshistorikkperiode>> {
         val sorterteVedtak = sorterVedtak(vedtaksliste, datoOpprettetPerBehandling)
-        return sorterteVedtak.fold(listOf<Pair<UUID, List<Vedtaksperiode>>>()) { acc, vedtak ->
+        return sorterteVedtak.fold(listOf<Pair<UUID, List<Vedtakshistorikkperiode>>>()) { acc, vedtak ->
             acc + Pair(vedtak.behandlingId, lagTotalbildeForNyttVedtak(vedtak, acc))
         }.toMap()
     }
 
-    private fun lagTotalbildeForNyttVedtak(vedtak: Vedtak,
-                                           acc: List<Pair<UUID, List<Vedtaksperiode>>>): List<Vedtaksperiode> {
-        return if (vedtak.resultatType == ResultatType.INNVILGE) {
-            val nyePerioder = vedtak.perioder?.perioder ?: error("Finner ikke perioder på vedtak=${vedtak.behandlingId}")
-            val førsteFraDato = nyePerioder.first().datoFra
-            avkortTidligerePerioder(acc.lastOrNull(), førsteFraDato) + nyePerioder
-        } else if (vedtak.resultatType == ResultatType.SANKSJONERE) {
-            splitOppPerioderSomErSanksjonert(acc, vedtak)
-        } else {
-            val opphørFom = vedtak.opphørFom ?: error("Mangler dato for opphør på vedtak=${vedtak.behandlingId}")
-            avkortTidligerePerioder(acc.lastOrNull(), opphørFom)
+    private fun lagTotalbildeForNyttVedtak(behandlingHistorikkData: AndelHistorikkBeregner.BehandlingHistorikkData,
+                                           acc: List<Pair<UUID, List<Vedtakshistorikkperiode>>>): List<Vedtakshistorikkperiode> {
+
+        val vedtak = behandlingHistorikkData.vedtakDto
+        return when (vedtak) {
+            is InnvilgelseOvergangsstønad -> {
+                val nyePerioder = vedtak.perioder.map { VedtakshistorikkperiodeOvergangsstønad(it) }
+                val førsteFraDato = nyePerioder.first().datoFra
+                avkortTidligerePerioder(acc.lastOrNull(), førsteFraDato) + nyePerioder
+            }
+            is InnvilgelseBarnetilsyn -> {
+                val perioder = beregningBarnetilsyn.beregnYtelseBarnetilsyn(vedtak).map { VedtakshistorikkperiodeBarnetilsyn(it, behandlingHistorikkData.aktivitetArbeid) }
```

Comment:
Tror BeregningBarnetilsynController kommer implementere en ny metode for å hente "beregnet beløp", der man slår sammen Tilkjent Ytelse og Vedtak, tenker den også burde tas i bruk her etterpå